watertap/tools/parameter_sweep.py

```diff
@@ -319,108 +342,290 @@ def _interp_nan_values(global_values, global_results):
-                                         ``model``, e.g.,
-                                         ``reinitialize_function(model, **reinitialize_kwargs)``.
-                                         The default uses no kwargs.
+    # Get the outputs from model
+    if solver_termination_condition == "optimal":
+        for key in output_dict['outputs'].keys():
+            outcome = model.find_component(key)
+            output_dict["outputs"][key]["value"][case_number] = pyo.value(outcome)
 
-        mpi_comm (optional) : User-provided MPI communicator for parallel parameter sweeps.
-                              If None COMM_WORLD will be used. The default is sufficient for most
-                              users.
+        # 1. Variables
+        # for var in variables_in_activated_equalities_set(model.fs):
+        #     output_dict["outputs"][var.name]["value"][case_number] = var.value
 
-        debugging_data_dir (optional) : Save results on a per-process basis for parallel debugging
-                                        purposes. If None no `debugging` data will be saved.
+        # # 2. Expressions
+        # for expr in expressions_set(model.fs):
+        #     output_dict["outputs"][expr.name]["value"][case_number] = pyo.value(expr)
 
-        interpolate_nan_outputs (optional) : When the parameter sweep has finished, interior values
-                                             of np.nan will be replaced with a value obtained via
-                                             a linear interpolation of their surrounding valid neighbors.
-                                             If true, a second output file with the extension "_clean"
-                                             will be saved alongside the raw (un-interpolated) values. 
+        # # 3. Objectives
+        # for obj in total_objectives_set(model):
+        #     output_dict["outputs"][obj.name]["value"][case_number] = pyo.value(obj)
 
-        num_samples (optional) : If the user is using sampling techniques rather than a linear grid
-                                 of values, they need to set the number of samples
+    else:
+        for key in output_dict['outputs'].keys():
+            outcome = model.find_component(key)
+            output_dict["outputs"][key]["value"][case_number] = np.nan
+
+        # 1. Variables: We will set everything to NaNs unless they are an input
+        # for var in variables_in_activated_equalities_set(model.fs):
+        #     if var.name in op_ps_dict.keys():
+        #         # This conditional exists so that we capture the actual input
+        #         # values. This is a redundancy since we already capture the
+        #         # inputs above.
+        #         output_dict["outputs"][var.name]["value"][case_number] = var.value
+        #     else:
+        #         output_dict["outputs"][var.name]["value"][case_number] = np.nan
```

Comment:
@k1nshuk do we need to save the sweep params in `output_dict["outputs"]` as well?

Reply:
I guess obtaining them from the Pyomo model is a good sanity check to confirm that you're changing model values as expected, but as far as storing them in the final output file, maybe not necessary.